soh/soh/Enhancements/cosmetics/NoMasterSword.cpp

```diff
@@ -0,0 +1,115 @@
+        }
+        for (int i = 62; i <= 150; i++) {
+            std::string patchName = "adultNoMS." + std::to_string(2) + "." + std::to_string(i);
+            ResourceMgr_UnpatchGfxByName(gLinkAdultMirrorShieldSwordAndSheathNearDL, patchName.c_str());
+        }
+        for (int i = 61; i <= 118; i++) {
+            std::string patchName = "adultNoMS." + std::to_string(3) + "." + std::to_string(i);
+            ResourceMgr_UnpatchGfxByName(gLinkAdultMirrorShieldSwordAndSheathFarDL, patchName.c_str());
+        }
+        ResourceMgr_UnpatchGfxByName(gLinkAdultHylianShieldSwordAndSheathNearDL, "adultNoMSHylianShield1");
+        ResourceMgr_UnpatchGfxByName(gLinkAdultHylianShieldSwordAndSheathFarDL, "adultNoMSHylianShield2");
+        ResourceMgr_UnpatchGfxByName(gLinkAdultMasterSwordAndSheathNearDL, "adultNoMasterSword1");
+        ResourceMgr_UnpatchGfxByName(gLinkAdultMasterSwordAndSheathFarDL, "adultNoMasterSword2");
+    }
+}
+
+// copied from z_player (right above Player_StartMode_TimeTravel)
+static Vec3f D_808546F4 = { -1.0f, 69.0f, 20.0f };
+
+void RegisterNoMasterSword() {
+    // don't show link taking out master sword to put in pedestal when we don't have a master sword
+    REGISTER_VB_SHOULD(VB_SHOW_MASTER_SWORD_TO_PLACE_IN_PEDESTAL, {
+        // if the vanilla condition is false respect it
+        if (!*should) {
+            return;
+        }
+
+        if (IS_RANDO && 
+            Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD) &&
+            !CHECK_OWNED_EQUIP(EQUIP_TYPE_SWORD, EQUIP_INV_SWORD_MASTER)) {
+            
+            *should = false;
+        }
+    });
+    
+    // skip post pedestal animation when we don't have a master sword
+    REGISTER_VB_SHOULD(VB_EXECUTE_PLAYER_STARTMODE_FUNC, {
+        int32_t startMode = va_arg(args, int32_t);
+        Player* player = GET_PLAYER(gPlayState);
+
+        if (startMode == PLAYER_START_MODE_TIME_TRAVEL && IS_RANDO && Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD) &&
+            !CHECK_OWNED_EQUIP(EQUIP_TYPE_SWORD, EQUIP_INV_SWORD_MASTER)) {
+            // don't run the vanilla startMode func
+            *should = false;
+            
+            // position link correctly
+            Math_Vec3f_Copy(&player->actor.world.pos, &D_808546F4);
+            player->yaw = player->actor.shape.rot.y = -0x8000;
+            
+            // execute the idle startMode func
+            Player_StartMode_Idle(gPlayState, player);
+        }
+    });
+
+    COND_HOOK(OnPlayerUpdate, true, []{
+        static uint16_t lastItemOnB = gSaveContext.equips.buttonItems[0];
+        if (lastItemOnB != gSaveContext.equips.buttonItems[0]) {
+            UpdateNoMSPatch();
+            lastItemOnB = gSaveContext.equips.buttonItems[0];
+        }
+    });
+
+    COND_HOOK(OnSceneSpawnActors, true, UpdateNoMSPatch);
+}
+
+static RegisterShipInitFunc initFuncAlways(RegisterNoMasterSword);
```

Comment:
You'll want to change this to `static RegisterShipInitFunc initFunc(RegisterNoMasterSword, { "IS_RANDO" });`

This way it gets registered and unregistered on save file load, instead of on boot and checkbox toggling. At that point any of the other IS_RANDO checks can also be removed.